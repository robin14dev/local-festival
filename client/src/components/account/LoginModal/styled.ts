import styled from "styled-components";

export const Backdrop = styled.div`
  z-index: 20;
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100vh;
  background: rgba(0, 0, 0, 0.4);

  animation-duration: 0.4s;

  animation-fill-mode: forwards;
`;

export const Container = styled.div`
  z-index: 100;
  position: fixed;
  width: 400px;
  height: 460px;
  display: flex;
  flex-direction: column;

  text-align: center;
  background: #fff;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.7);
  border-radius: 10px;
  margin-top: -256px;
  margin-left: -200px;
  left: 50%;
  top: 50%;

  animation-duration: 0.4s;

  animation-fill-mode: forwards;

  & > h1 {
    margin: 2rem 0;
    font-size: 3rem;
    cursor: pointer;
    font-style: italic;
    font-family: "HS-Regular";
    color: var(--mainColor);
  }

  form {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 2rem 0;
    input {
      border-radius: 6px;
      border: 1.8px solid #b1aeae;
      padding: 0.5rem;
      width: 80%;
      height: 3rem;
      font-family: "NanumSquareRound";
      &::placeholder {
        opacity: 0.5;
      }
    }
    div {
      min-height: 1rem;
      color: red;
      font-size: 0.8rem;
      align-self: flex-start;
      margin-left: 3rem;
      margin-bottom: 1rem;
      padding-top: 0.2rem;
    }
    button {
      width: 80%;
      height: 3rem;
      margin-top: 1rem;
      font-size: 1rem;

      color: #fff;
      font-family: "NanumSquareRound";
      font-weight: bold;
      background: var(--mainColor);
      border-radius: 6px;
      cursor: pointer;
      transition : all 0.7s;
      &:hover {
        filter : brightness(1.15)
      }
      &:disabled {
        filter : brightness(0.8);
       
      }
    }
  }

  .footer {
    display: flex;
    /* background-color: yellow; */
    justify-content: center;
    align-items: center;
    font-size: 0.9rem;
    font-weight: 500;

    button {
      font-family: "NanumSquareRound";
      margin-left: 1rem;
      font-size: 1rem;
      /* background-color: yellowgreen; */
      color: var(--primaryPurple);
      font-weight: bold;
      padding-bottom: 3px;
      transition: all 0.2s;
      &:hover {
        color: var(--primaryOrange);
        border-radius: 5px;
      }
    }
  }

  p {
    color: red;
  }

  .modalClose {
    display: none;
  }

  @media screen and (max-width: 428px) {
    width: 100%;
    height: 100%;
    border-radius: 0;
    margin: 0;
    left: 0;
    top: 0;

    .modalClose {
      display: block;
      margin-top: 2rem;
      color: gray;
      font-family: "NanumSquareRound";
    }
  }
`;
// export const Container = styled.div<{ isHide: boolean }>`
//   z-index: 100;
//   position: fixed;
//   width: 400px;
//   height: 460px;
//   display: flex;
//   flex-direction: column;

//   text-align: center;
//   background: #fff;
//   box-shadow: 0 2px 8px rgba(0, 0, 0, 0.7);
//   border-radius: 10px;
//   margin-top: -256px;
//   margin-left: -200px;
//   left: 50%;
//   top: 50%;

//   animation-duration: 0.4s;
//   animation-name: ${(props) => (props.isHide ? "slide-up" : "slide-down")};
//   animation-fill-mode: forwards;

//   & > h1 {
//     margin: 2rem 0;
//     font-size: 3rem;
//     cursor: pointer;
//     font-style: italic;
//     font-family: "HS-Regular";
//     color: var(--mainColor);
//   }

//   form {
//     display: flex;
//     flex-direction: column;
//     align-items: center;
//     margin: 2rem 0;
//     input {
//       border-radius: 6px;
//       border: 1.8px solid #b1aeae;
//       padding: 0.5rem;
//       width: 80%;
//       height: 3rem;
//       font-family: "NanumSquareRound";
//       &::placeholder {
//         opacity: 0.5;
//       }
//     }
//     div {
//       min-height: 1rem;
//       color: red;
//       font-size: 0.8rem;
//       align-self: flex-start;
//       margin-left: 3rem;
//       margin-bottom: 1rem;
//       padding-top: 0.2rem;
//     }
//     button {
//       width: 80%;
//       height: 3rem;
//       margin-top: 1rem;
//       font-size: 1rem;

//       color: #fff;
//       font-family: "NanumSquareRound";
//       font-weight: bold;
//       background: var(--mainColor);
//       border-radius: 6px;
//       cursor: pointer;
//     }
//   }

//   .footer {
//     display: flex;
//     /* background-color: yellow; */
//     justify-content: center;
//     align-items: center;
//     font-size: 0.9rem;
//     font-weight: 500;

//     button {
//       font-family: "NanumSquareRound";
//       margin-left: 1rem;
//       font-size: 1rem;
//       /* background-color: yellowgreen; */
//       color: var(--primaryPurple);
//       font-weight: bold;
//       padding-bottom: 3px;
//       transition: all 0.2s;
//       &:hover {
//         color: var(--primaryOrange);
//         border-radius: 5px;
//       }
//     }
//   }

//   p {
//     color: red;
//   }

//   .modalClose {
//     display: none;
//   }

//   @media screen and (max-width: 428px) {
//     width: 100%;
//     height: 100%;
//     border-radius: 0;
//     margin: 0;
//     left: 0;
//     top: 0;

//     .modalClose {
//       display: block;
//       margin-top: 2rem;
//       color: gray;
//       font-family: "NanumSquareRound";
//     }
//   }
// `;
