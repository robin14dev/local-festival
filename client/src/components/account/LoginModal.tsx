import axios, { AxiosError } from "axios";
import React, { useCallback, useContext, useState } from "react";
import styled from "styled-components";
import Signup from "./Signup";
import { UserContext } from "../../contexts/userContext";

const Backdrop = styled.div<{ isHide: boolean }>`
  z-index: 20;
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100vh;
  background: rgba(0, 0, 0, 0.4);

  animation-duration: 0.4s;
  animation-name: ${(props) => (props.isHide ? "bright-soft" : "dark-soft")};
  animation-fill-mode: forwards;
`;
const Container = styled.div<{ isHide: boolean }>`
  z-index: 100;
  position: fixed;
  width: 400px;
  height: 460px;
  display: flex;
  flex-direction: column;

  text-align: center;
  background: #fff;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.7);
  border-radius: 10px;
  margin-top: -256px;
  margin-left: -200px;
  left: 50%;
  top: 50%;

  animation-duration: 0.4s;
  animation-name: ${(props) => (props.isHide ? "slide-up" : "slide-down")};
  animation-fill-mode: forwards;

  & > h1 {
    margin: 2rem 0;
    font-size: 3rem;
    cursor: pointer;
    font-style: italic;
    font-family: "HS-Regular";
    color: var(--mainColor);
  }

  form {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 2rem 0;
    input {
      border-radius: 6px;
      border: 1.8px solid #b1aeae;
      padding: 0.5rem;
      width: 80%;
      height: 3rem;
      font-family: "NanumSquareRound";
      &::placeholder {
        opacity: 0.5;
      }
    }
    div {
      min-height: 1rem;
      color: red;
      font-size: 0.8rem;
      align-self: flex-start;
      margin-left: 3rem;
      margin-bottom: 1rem;
      padding-top: 0.2rem;
    }
    button {
      width: 80%;
      height: 3rem;
      margin-top: 1rem;
      font-size: 1rem;

      color: #fff;
      font-family: "NanumSquareRound";
      font-weight: bold;
      background: var(--mainColor);
      border-radius: 6px;
      cursor: pointer;
    }
  }

  .footer {
    display: flex;
    /* background-color: yellow; */
    justify-content: center;
    align-items: center;
    font-size: 0.9rem;
    font-weight: 500;

    button {
      font-family: "NanumSquareRound";
      margin-left: 1rem;
      font-size: 1rem;
      /* background-color: yellowgreen; */
      color: var(--primaryPurple);
      font-weight: bold;
      padding-bottom: 3px;
      transition: all 0.2s;
      &:hover {
        color: var(--primaryOrange);
        border-radius: 5px;
      }
    }
  }

  p {
    color: red;
  }

  .modalClose {
    display: none;
  }

  @media screen and (max-width: 428px) {
    width: 100%;
    height: 100%;
    border-radius: 0;
    margin: 0;
    left: 0;
    top: 0;

    .modalClose {
      display: block;
      margin-top: 2rem;
      color: gray;
      font-family: "NanumSquareRound";
    }
  }
`;

/*
백드롭은 계속 어둡게 하고 
로그인  => 회원가입 => 성공 => 실패

*/

type LoginProps = {
  setIsLoginModal: React.Dispatch<React.SetStateAction<boolean>>;
};

const LoginModal = ({ setIsLoginModal }: LoginProps) => {
  console.log("logimModal!!!");

  const { setAuthState } = useContext(UserContext);
  const [userInfo, setUserInfo] = useState({ account: "", password: "" });
  const { account, password } = userInfo;
  const [errMessage, setErrMessage] = useState("");
  const [isHide, setIsHide] = useState(false);
  const [isLogin, setIsLogin] = useState(true);
  const [isSignup, setIsSignup] = useState(false);

  const errMessages = [
    "사용자가 존재하지 않습니다",
    "비밀번호가 일치하지 않습니다",
  ];

  const handleUserInfo = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => {
      if (e.target.value.length > 0) setErrMessage("");
      setUserInfo((prevInfo) => ({
        ...prevInfo,
        [e.target.name]: e.target.value,
      }));
    },
    []
  );
  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
    try {
      e.preventDefault();
      const response = await axios.post(
        `${process.env.REACT_APP_SERVER_URL}/users/signin`,
        {
          account: userInfo.account,
          password: userInfo.password,
        }
      );
      const { nickname, userId, account, defaultPic } = response.data.info;
      const user = {
        account,
        nickname,
        userId,
        defaultPic,
        loginStatus: true,
      };
      setAuthState(user);
      sessionStorage.setItem("user", JSON.stringify(user));
      sessionStorage.setItem("accesstoken", response.data.info.token);

      /**
       * loginHandler 안에 로그인 성공시, 찜 데이터를 가져 와야 하는데 분리해 주어함
       *
       * 로그인버튼 누른다
       * 로그인을 성공해서 유저정보를 가져온다.
       * 가져온 유저정보로 authState를 업데이트 한다
       * 해당 유저정보로 다시 찜 정보를 가져오기 위해 pick api를 보낸다.
       */

      modalClose();
    } catch (err: unknown) {
      if (err instanceof AxiosError) {
        console.log(err.response?.data.message);
        if (
          err.response?.data.message ===
          "Wrong account And Password Combination"
        ) {
          setErrMessage(errMessages[1]);
        } else if (err.response?.data.message === "User Doesn't Exist") {
          setErrMessage(errMessages[0]);
        } else {
          console.log("그밖에에러");
        }
      } else {
        console.log(err);
      }
    }
  };
  const modalClose = () => {
    setIsHide(true);
    setIsSignup(false);
    setTimeout(() => {
      setIsLoginModal(false);
    }, 400);
  };
  return (
    <>
      <Backdrop onClick={modalClose} isHide={isHide}>
        {isLogin && (
          <Container
            data-testid="LoginModal"
            isHide={isHide}
            onClick={(e) => {
              e.stopPropagation();
            }}
          >
            <h1>Loco</h1>
            <form onSubmit={handleSubmit}>
              <input
                name="account"
                placeholder="email을 입력해 주세요"
                onChange={handleUserInfo}
                type="text"
                value={account}
                required
              ></input>
              <div>
                {errMessage === "사용자가 존재하지 않습니다" && errMessage}
              </div>
              <input
                name="password"
                placeholder="비밀번호를 입력해 주세요"
                onChange={handleUserInfo}
                type="password"
                value={password}
                required
              ></input>
              <div>
                {errMessage === "비밀번호가 일치하지 않습니다" && errMessage}
              </div>
              <button type="submit">로그인</button>
            </form>
            <div className="footer">
              <span>아직 계정이 없으신가요?</span>
              <button
                onClick={() => {
                  setIsLogin(false);
                  setIsSignup(true);
                }}
              >
                회원가입
              </button>
            </div>
            <button
              className="modalClose"
              onClick={() => {
                setIsLoginModal(false);
              }}
            >
              돌아가기
            </button>
          </Container>
        )}

        {isSignup && (
          <Signup setIsSignup={setIsSignup} setIsLogin={setIsLogin} />
        )}
      </Backdrop>
    </>
  );
};

export default LoginModal;
