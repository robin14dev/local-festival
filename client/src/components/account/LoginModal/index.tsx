import React, { useCallback, useContext, useState } from 'react'
import axios, { AxiosError } from 'axios'
import { UserContext } from 'contexts/userContext'
import { ModalDispatchContext } from 'contexts/ModalContext'
import { Backdrop, Container } from './styled'

type ResponseDataInfo = {
    nickname: string
    userId: number
    account: string
    defaultPic: string
    token: string
}

const errMessages = {
    inValidUser: '사용자가 존재하지 않습니다',
    inValidPassword: '비밀번호가 일치하지 않습니다',
}

const LoginModal = () => {
    const setModalKey = useContext(ModalDispatchContext)
    const { setAuthState } = useContext(UserContext)
    const [userInfo, setUserInfo] = useState({ account: '', password: '' })
    const { account, password } = userInfo
    const [errMessage, setErrMessage] = useState('')

    const handleUserInfo = useCallback(
        (e: React.ChangeEvent<HTMLInputElement>) => {
            if (e.target.value.length > 0) setErrMessage('')
            setUserInfo((prevInfo) => ({
                ...prevInfo,
                [e.target.name]: e.target.value,
            }))
        },
        []
    )
    const login = async (account: string, password: string) => {
        try {
            const response = await axios.post(
                `${process.env.REACT_APP_SERVER_URL}/users/signin`,
                {
                    account,
                    password,
                }
            )
            const {
                nickname,
                userId,
                account: accountName,
                defaultPic,
                token,
            }: ResponseDataInfo = response.data.info
            const user = {
                account: accountName,
                nickname,
                userId,
                defaultPic,
                loginStatus: true,
            }
            setAuthState(user)
            sessionStorage.setItem('user', JSON.stringify(user))
            sessionStorage.setItem('accesstoken', token)
            modalClose()
        } catch (err: unknown) {
            if (err instanceof AxiosError) {
                console.log(err.response?.data.message)
                if (
                    err.response?.data.message ===
                    'Wrong account And Password Combination'
                ) {
                    setErrMessage(errMessages.inValidPassword)
                } else if (
                    err.response?.data.message === "User Doesn't Exist"
                ) {
                    setErrMessage(errMessages.inValidUser)
                } else {
                    console.log('그밖에에러')
                }
            } else {
                console.log(err)
            }
        }
    }
    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault()
        login(account, password)
    }
    const modalClose = () => {
        setModalKey('')
    }
    const openSignupModal = () => {
        setModalKey('SignupModal')
    }

    return (
        <Backdrop onClick={modalClose}>
            <Container
                data-testid="LoginModal"
                onClick={(e) => {
                    e.stopPropagation()
                }}
            >
                <h1>Loco</h1>
                <form onSubmit={handleSubmit}>
                    <input
                        autoFocus
                        name="account"
                        placeholder="email을 입력해 주세요"
                        onChange={handleUserInfo}
                        type="text"
                        value={account}
                        required
                    ></input>
                    <div>
                        {errMessage === '사용자가 존재하지 않습니다' &&
                            errMessage}
                    </div>
                    <input
                        name="password"
                        placeholder="비밀번호를 입력해 주세요"
                        onChange={handleUserInfo}
                        type="password"
                        value={password}
                        required
                    ></input>
                    <div>
                        {errMessage === '비밀번호가 일치하지 않습니다' &&
                            errMessage}
                    </div>
                    <button disabled={!account || !password}>로그인</button>
                </form>
                <div className="footer">
                    <span>아직 계정이 없으신가요?</span>
                    <button type="button" onClick={openSignupModal}>
                        회원가입
                    </button>
                </div>
            </Container>
        </Backdrop>
    )
}

export default LoginModal
