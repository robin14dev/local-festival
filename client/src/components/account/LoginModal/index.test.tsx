import React, { useContext } from 'react'
import {
    render,
    screen,
    waitFor,
    waitForElementToBeRemoved,
} from '@testing-library/react'
import LoginModal from '.'
import userEvent from '@testing-library/user-event'
import {
    ModalContextProvider,
    ModalDispatchContext,
} from 'contexts/ModalContext'
import {
    interceptLoginFailureOnPassword,
    interceptLoginFailureOnUserId,
    interceptLoginSuccess,
} from 'test/mocks/mockServer/scope/user'

describe('<LoginModal />', () => {
    test('컴포넌트가 성공적으로 렌더링 됩니다.', () => {
        const { container } = render(<LoginModal />)

        const emailInput = screen.getByPlaceholderText('email을 입력해 주세요')
        const passwordInput =
            screen.getByPlaceholderText('비밀번호를 입력해 주세요')
        const submitButton = screen.getByRole('button', { name: '로그인' })
        const signupButton = screen.getByRole('button', { name: '회원가입' })
        expect(emailInput).toBeInTheDocument()
        expect(passwordInput).toBeInTheDocument()
        expect(submitButton).toBeInTheDocument()
        expect(signupButton).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })
    test('인풋에 입력한 텍스트 값이 그대로 렌더링 됩니다', async () => {
        const user = userEvent.setup()
        const { container } = render(<LoginModal />)

        const emailInput = screen.getByPlaceholderText('email을 입력해 주세요')
        const passwordInput =
            screen.getByPlaceholderText('비밀번호를 입력해 주세요')

        expect(emailInput).toHaveValue('')
        expect(passwordInput).toHaveValue('')

        const emailText = 'account@mail.com'
        const passwordText = '1234'
        await user.click(emailInput)
        await user.keyboard(emailText)
        expect(emailInput).toHaveValue(emailText)
        await user.tab()
        await user.keyboard(passwordText)
        expect(passwordInput).toHaveValue(passwordText)
        expect(container).toMatchSnapshot()
    })
    test('인풋 중 하나라도 비어있으면 로그인 버튼이 비활성화 됩니다.', async () => {
        const user = userEvent.setup()
        const { container } = render(<LoginModal />)

        const emailInput = screen.getByPlaceholderText('email을 입력해 주세요')
        const passwordInput =
            screen.getByPlaceholderText('비밀번호를 입력해 주세요')

        expect(emailInput).toHaveValue('')
        expect(passwordInput).toHaveValue('')

        const emailText = 'account@mail.com'
        const passwordText = '1234'
        const testCases = [
            { input: emailInput, text: emailText },
            { input: passwordInput, text: passwordText },
        ]

        const submitButton = screen.getByRole('button', { name: '로그인' })
        expect(submitButton).toBeInTheDocument()
        for (const testCase of testCases) {
            const { input, text } = testCase
            await user.click(input)
            await user.keyboard(text)
            expect(input).toHaveValue(text)
            expect(submitButton).toBeDisabled()
            expect(container).toMatchSnapshot()
            await user.clear(input)
        }
    })
    test('회원가입 버튼을 누르면 로그인 모달이 사라지고 회원가입 모달이 렌더링 됩니다', async () => {
        const user = userEvent.setup()
        const TestComponent = (): JSX.Element => {
            const setModalKey = useContext(ModalDispatchContext)
            const showLoginModal = () => {
                setModalKey('LoginModal')
            }
            return <button onClick={showLoginModal}>loginModal</button>
        }
        const { container } = render(
            <ModalContextProvider>
                <TestComponent />
            </ModalContextProvider>
        )

        const showLoginModalButton = screen.getByRole('button', {
            name: 'loginModal',
        })
        expect(showLoginModalButton).toBeInTheDocument()
        await user.click(showLoginModalButton)

        const loginModal = screen.getByTestId('LoginModal')
        expect(loginModal).toBeInTheDocument()

        const signupButton = screen.getByRole('button', { name: '회원가입' })
        expect(signupButton).toBeInTheDocument()
        await user.click(signupButton)
        const SignupModal = screen.getByTestId('SignupModal')
        expect(SignupModal).toBeInTheDocument()
        expect(loginModal).not.toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })
    describe('[로그인 실패 시]', () => {
        test('이메일이 존재하지 않을 때 에러 메시지를 렌더링 합니다.', async () => {
            interceptLoginFailureOnUserId()

            const user = userEvent.setup()
            const { container } = render(<LoginModal />)
            const emailInput =
                screen.getByPlaceholderText('email을 입력해 주세요')
            const submitButton = screen.getByRole('button', { name: '로그인' })

            await user.click(emailInput)
            await user.keyboard('user1')
            await user.tab()
            await user.keyboard('password1')
            await user.click(submitButton)
            const errorMessage =
                await screen.findByText('사용자가 존재하지 않습니다')
            expect(errorMessage).toBeInTheDocument()
            expect(container).toMatchSnapshot()
        })
        test('잘못된 비밀번호 입력시, 에러메시지를 렌더링 합니다.', async () => {
            interceptLoginFailureOnPassword()

            const user = userEvent.setup()
            const { container } = render(<LoginModal />)
            const emailInput =
                screen.getByPlaceholderText('email을 입력해 주세요')
            const submitButton = screen.getByRole('button', { name: '로그인' })

            await user.click(emailInput)
            await user.keyboard('user1')
            await user.tab()
            await user.keyboard('password1')
            await user.click(submitButton)
            const errorMessage =
                await screen.findByText('비밀번호가 일치하지 않습니다')
            expect(errorMessage).toBeInTheDocument()
            expect(container).toMatchSnapshot()
        })
    })
    describe('[로그인 성공 시]', () => {
        test('로그인 모달이 사라집니다.', async () => {
            interceptLoginSuccess()
            const user = userEvent.setup()

            const TestComponent = (): JSX.Element => {
                const setModalKey = useContext(ModalDispatchContext)
                const showLoginModal = () => {
                    setModalKey('LoginModal')
                }
                return (
                    <div>
                        <h1>TestComponent</h1>
                        <button onClick={showLoginModal}>loginModal</button>
                    </div>
                )
            }

            const { container } = render(
                <ModalContextProvider>
                    <TestComponent />
                </ModalContextProvider>
            )

            const showLoginModalButton = screen.getByRole('button', {
                name: 'loginModal',
            })
            expect(showLoginModalButton).toBeInTheDocument()
            await user.click(showLoginModalButton)

            const loginModal = screen.getByTestId('LoginModal')
            expect(loginModal).toBeInTheDocument()

            const emailInput =
                screen.getByPlaceholderText('email을 입력해 주세요')
            const submitButton = screen.getByRole('button', { name: '로그인' })

            await user.click(emailInput)
            await user.keyboard('user1')
            await user.tab()
            await user.keyboard('password1')
            await user.click(submitButton)

            await waitForElementToBeRemoved(loginModal)
            expect(container).toMatchSnapshot()

            //    await waitFor(() => { // ! await이 붙어야 함

            //     expect(showLoginModalButton).toBeInTheDocument()
            //     expect(container).toMatchSnapshot()
            //     })
        })
    })
})
