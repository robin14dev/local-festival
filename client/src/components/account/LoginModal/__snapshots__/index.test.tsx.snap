// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginModal /> [로그인 성공 시] 로그인 모달이 사라집니다. 1`] = `
<div>
  <div>
    <h1>
      TestComponent
    </h1>
    <button>
      loginModal
    </button>
  </div>
</div>
`;

exports[`<LoginModal /> [로그인 실패 시] 이메일이 존재하지 않을 때 에러 메시지를 렌더링 합니다. 1`] = `
<div>
  <div
    class="sc-gKPRtg cxiGOt"
  >
    <div
      class="sc-iBYQkv hHEDdX"
      data-testid="LoginModal"
    >
      <h1>
        Loco
      </h1>
      <form>
        <input
          name="account"
          placeholder="email을 입력해 주세요"
          required=""
          type="text"
          value="user1"
        />
        <div>
          사용자가 존재하지 않습니다
        </div>
        <input
          name="password"
          placeholder="비밀번호를 입력해 주세요"
          required=""
          type="password"
          value="password1"
        />
        <div />
        <button>
          로그인
        </button>
      </form>
      <div
        class="footer"
      >
        <span>
          아직 계정이 없으신가요?
        </span>
        <button
          type="button"
        >
          회원가입
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<LoginModal /> [로그인 실패 시] 잘못된 비밀번호 입력시, 에러메시지를 렌더링 합니다. 1`] = `
<div>
  <div
    class="sc-gKPRtg cxiGOt"
  >
    <div
      class="sc-iBYQkv hHEDdX"
      data-testid="LoginModal"
    >
      <h1>
        Loco
      </h1>
      <form>
        <input
          name="account"
          placeholder="email을 입력해 주세요"
          required=""
          type="text"
          value="user1"
        />
        <div />
        <input
          name="password"
          placeholder="비밀번호를 입력해 주세요"
          required=""
          type="password"
          value="password1"
        />
        <div>
          비밀번호가 일치하지 않습니다
        </div>
        <button>
          로그인
        </button>
      </form>
      <div
        class="footer"
      >
        <span>
          아직 계정이 없으신가요?
        </span>
        <button
          type="button"
        >
          회원가입
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<LoginModal /> 인풋 중 하나라도 비어있으면 로그인 버튼이 비활성화 됩니다. 1`] = `
<div>
  <div
    class="sc-gKPRtg cxiGOt"
  >
    <div
      class="sc-iBYQkv hHEDdX"
      data-testid="LoginModal"
    >
      <h1>
        Loco
      </h1>
      <form>
        <input
          name="account"
          placeholder="email을 입력해 주세요"
          required=""
          type="text"
          value="account@mail.com"
        />
        <div />
        <input
          name="password"
          placeholder="비밀번호를 입력해 주세요"
          required=""
          type="password"
          value=""
        />
        <div />
        <button
          disabled=""
        >
          로그인
        </button>
      </form>
      <div
        class="footer"
      >
        <span>
          아직 계정이 없으신가요?
        </span>
        <button
          type="button"
        >
          회원가입
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<LoginModal /> 인풋 중 하나라도 비어있으면 로그인 버튼이 비활성화 됩니다. 2`] = `
<div>
  <div
    class="sc-gKPRtg cxiGOt"
  >
    <div
      class="sc-iBYQkv hHEDdX"
      data-testid="LoginModal"
    >
      <h1>
        Loco
      </h1>
      <form>
        <input
          name="account"
          placeholder="email을 입력해 주세요"
          required=""
          type="text"
          value=""
        />
        <div />
        <input
          name="password"
          placeholder="비밀번호를 입력해 주세요"
          required=""
          type="password"
          value="1234"
        />
        <div />
        <button
          disabled=""
        >
          로그인
        </button>
      </form>
      <div
        class="footer"
      >
        <span>
          아직 계정이 없으신가요?
        </span>
        <button
          type="button"
        >
          회원가입
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<LoginModal /> 인풋에 입력한 텍스트 값이 그대로 렌더링 됩니다 1`] = `
<div>
  <div
    class="sc-gKPRtg cxiGOt"
  >
    <div
      class="sc-iBYQkv hHEDdX"
      data-testid="LoginModal"
    >
      <h1>
        Loco
      </h1>
      <form>
        <input
          name="account"
          placeholder="email을 입력해 주세요"
          required=""
          type="text"
          value="account@mail.com"
        />
        <div />
        <input
          name="password"
          placeholder="비밀번호를 입력해 주세요"
          required=""
          type="password"
          value="1234"
        />
        <div />
        <button>
          로그인
        </button>
      </form>
      <div
        class="footer"
      >
        <span>
          아직 계정이 없으신가요?
        </span>
        <button
          type="button"
        >
          회원가입
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<LoginModal /> 컴포넌트가 성공적으로 렌더링 됩니다. 1`] = `
<div>
  <div
    class="sc-gKPRtg cxiGOt"
  >
    <div
      class="sc-iBYQkv hHEDdX"
      data-testid="LoginModal"
    >
      <h1>
        Loco
      </h1>
      <form>
        <input
          name="account"
          placeholder="email을 입력해 주세요"
          required=""
          type="text"
          value=""
        />
        <div />
        <input
          name="password"
          placeholder="비밀번호를 입력해 주세요"
          required=""
          type="password"
          value=""
        />
        <div />
        <button
          disabled=""
        >
          로그인
        </button>
      </form>
      <div
        class="footer"
      >
        <span>
          아직 계정이 없으신가요?
        </span>
        <button
          type="button"
        >
          회원가입
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<LoginModal /> 회원가입 버튼을 누르면 로그인 모달이 사라지고 회원가입 모달이 렌더링 됩니다 1`] = `
<div>
  <div
    class="sc-gswNZR fmubx"
  >
    <div
      class="sc-dkrFOg ggjaNf"
      data-testid="SignupModal"
    >
      <div
        class="signupForm"
      >
        <h1>
          LoCo
        </h1>
        <form>
          <div
            class="validInfo"
          >
            <label
              for="account"
            >
              이메일 주소
            </label>
             
            <div
              class="sc-jSUZER gIZaff"
            />
            <button
              disabled=""
              name="account"
            >
              중복 확인
            </button>
          </div>
          <input
            name="account"
            required=""
            type="text"
            value=""
          />
          <div
            class="validInfo"
          >
            <label
              for="nickname"
            >
              닉네임
            </label>
            <div
              class="sc-jSUZER gIZaff"
            />
            <button
              disabled=""
              name="nickname"
            >
              중복 확인
            </button>
          </div>
          <input
            name="nickname"
            required=""
            type="text"
            value=""
          />
          <div
            class="validInfo"
          >
            <label
              for="password"
            >
              비밀번호
            </label>
             
            <div
              class="sc-jSUZER gIZaff"
            />
          </div>
          <input
            name="password"
            required=""
            type="password"
            value=""
          />
          <div
            class="validInfo"
          >
            <label
              for="passwordCheck"
            >
              비밀번호 확인
            </label>
            <div
              class="sc-jSUZER gIZaff"
            />
          </div>
          <input
            name="passwordCheck"
            required=""
            type="password"
            value=""
          />
          <button>
            회원가입
          </button>
        </form>
        <div
          class="sc-eDvSVe brirrE"
        >
          <p>
            이미 계정이 있으신가요?
          </p>
          <button>
            로그인
          </button>
        </div>
        <button
          class="modalClose"
        >
          돌아가기
        </button>
      </div>
    </div>
  </div>
  <button>
    loginModal
  </button>
</div>
`;
